### ISSUE TITLE
Exception when syncing from Shopify to Postgres/BQ with normalization

### ISSUE THREAD

**Issue by sherifnada** [airbytehq/airbyte#1904]
## Expected Behavior
A user reported the following exception when syncing from shopify to BQ and Postgres

```
2021-01-31 14:54:06 INFO (/tmp/workspace/7/0) DefaultSyncWorker(run):110 - Running normalization.
2021-01-31 14:54:06 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Checking if airbyte/normalization:0.1.10 exists...
2021-01-31 14:54:06 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - airbyte/normalization:0.1.10 not found locally. Attempting to pull the image...
2021-01-31 14:54:20 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Pulled airbyte/normalization:0.1.10 from remote.
2021-01-31 14:54:20 DEBUG (/tmp/workspace/7/0) DockerProcessBuilderFactory(create):104 - Preparing command: docker run --rm -i -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -w /data/7/0/normalize --network host airbyte/normalization:0.1.10 run --integration-type bigquery --config target_config.json --catalog catalog.json
2021-01-31 14:54:25 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Namespace(config='target_config.json', integration_type=<DestinationType.bigquery: 'bigquery'>, out='/data/7/0/normalize')
2021-01-31 14:54:25 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - transform_bigquery
2021-01-31 14:54:25 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Processing catalog.json...
2021-01-31 14:54:25 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Generating abandoned_checkouts.sql in /data/7/0/normalize/models/generated/
2021-01-31 14:54:25 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Generating collects.sql in /data/7/0/normalize/models/generated/
2021-01-31 14:54:25 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Generating custom_collections.sql in /data/7/0/normalize/models/generated/
2021-01-31 14:54:25 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Generating customers.sql in /data/7/0/normalize/models/generated/
2021-01-31 14:54:25 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Generating metafields.sql in /data/7/0/normalize/models/generated/
2021-01-31 14:54:25 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Generating order_refunds.sql in /data/7/0/normalize/models/generated/
2021-01-31 14:54:25 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Generating orders.sql in /data/7/0/normalize/models/generated/
2021-01-31 14:54:25 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Generating products.sql in /data/7/0/normalize/models/generated/
2021-01-31 14:54:25 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Generating transactions.sql in /data/7/0/normalize/models/generated/
2021-01-31 14:54:26 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Running with dbt=0.18.1
2021-01-31 14:54:28 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Found 9 models, 0 tests, 0 snapshots, 0 analyses, 325 macros, 0 operations, 0 seed files, 9 sources
2021-01-31 14:54:28 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 -
2021-01-31 14:54:28 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:28 | Concurrency: 32 threads (target='prod')
2021-01-31 14:54:28 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:28 |
2021-01-31 14:54:28 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:28 | 1 of 9 START table model earthly_shopify.abandoned_checkouts................................................. [RUN]
2021-01-31 14:54:28 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:28 | 3 of 9 START table model earthly_shopify.custom_collections.................................................. [RUN]
2021-01-31 14:54:28 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:28 | 2 of 9 START table model earthly_shopify.collects............................................................ [RUN]
2021-01-31 14:54:28 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:28 | 5 of 9 START table model earthly_shopify.metafields.......................................................... [RUN]
2021-01-31 14:54:28 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:28 | 7 of 9 START table model earthly_shopify.orders.............................................................. [RUN]
2021-01-31 14:54:28 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:28 | 8 of 9 START table model earthly_shopify.products............................................................ [RUN]
2021-01-31 14:54:28 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:28 | 4 of 9 START table model earthly_shopify.customers........................................................... [RUN]
2021-01-31 14:54:29 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:28 | 9 of 9 START table model earthly_shopify.transactions........................................................ [RUN]
2021-01-31 14:54:29 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:28 | 6 of 9 START table model earthly_shopify.order_refunds....................................................... [RUN]
2021-01-31 14:54:32 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:32 | 1 of 9 OK created table model earthly_shopify.abandoned_checkouts............................................ [[32mCREATE TABLE (584.0 rows, 1.9 MB processed)[0m in 3.06s]
2021-01-31 14:54:32 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:32 | 8 of 9 OK created table model earthly_shopify.products....................................................... [[32mCREATE TABLE (4.0 rows, 29.2 KB processed)[0m in 3.05s]
2021-01-31 14:54:32 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:32 | 3 of 9 OK created table model earthly_shopify.custom_collections............................................. [[32mCREATE TABLE (0.0 rows, 0.0 Bytes processed)[0m in 3.34s]
2021-01-31 14:54:32 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:32 | 2 of 9 OK created table model earthly_shopify.collects....................................................... [[32mCREATE TABLE (0.0 rows, 0.0 Bytes processed)[0m in 3.36s]
2021-01-31 14:54:32 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:32 | 5 of 9 ERROR creating table model earthly_shopify.metafields................................................. [[31mERROR[0m in 3.62s]
2021-01-31 14:54:32 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:32 | 6 of 9 OK created table model earthly_shopify.order_refunds.................................................. [[32mCREATE TABLE (644.0 rows, 996.7 KB processed)[0m in 3.84s]
2021-01-31 14:54:33 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:33 | 4 of 9 OK created table model earthly_shopify.customers...................................................... [[32mCREATE TABLE (15.7k rows, 18.0 MB processed)[0m in 4.90s]
2021-01-31 14:54:35 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:35 | 9 of 9 OK created table model earthly_shopify.transactions................................................... [[32mCREATE TABLE (16.4k rows, 30.4 MB processed)[0m in 6.46s]
2021-01-31 14:54:53 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:53 | 7 of 9 OK created table model earthly_shopify.orders......................................................... [[32mCREATE TABLE (15.6k rows, 119.0 MB processed)[0m in 24.93s]
2021-01-31 14:54:53 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:53 |
2021-01-31 14:54:53 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - 14:54:53 | Finished running 9 table models in 25.37s.
2021-01-31 14:54:54 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 -
2021-01-31 14:54:54 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - [31mCompleted with 1 error and 0 warnings:[0m
2021-01-31 14:54:54 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 -
2021-01-31 14:54:54 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - [33mDatabase Error in model metafields (models/generated/metafields.sql)[0m
2021-01-31 14:54:54 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Bad double value: true
2021-01-31 14:54:54 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - compiled SQL at ../build/run/airbyte_utils/models/generated/metafields.sql
2021-01-31 14:54:54 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 -
2021-01-31 14:54:54 INFO (/tmp/workspace/7/0) LineGobbler(voidCall):69 - Done. PASS=8 WARN=0 ERROR=1 SKIP=0 TOTAL=9
2021-01-31 14:54:54 DEBUG (/tmp/workspace/7/0) DefaultNormalizationRunner(close):97 - Closing tap process
2021-01-31 14:54:54 ERROR (/tmp/workspace/7/0) DefaultSyncWorker(run):117 - Normalization Failed.
io.airbyte.workers.WorkerException: Normalization Failed.
at io.airbyte.workers.DefaultSyncWorker.run(DefaultSyncWorker.java:114) [io.airbyte-airbyte-workers-0.14.1-alpha.jar:?]
at io.airbyte.workers.DefaultSyncWorker.run(DefaultSyncWorker.java:48) [io.airbyte-airbyte-workers-0.14.1-alpha.jar:?]
at io.airbyte.workers.wrappers.OutputConvertingWorker.run(OutputConvertingWorker.java:44) [io.airbyte-airbyte-workers-0.14.1-alpha.jar:?]
at io.airbyte.workers.wrappers.JobOutputSyncWorker.run(JobOutputSyncWorker.java:32) [io.airbyte-airbyte-workers-0.14.1-alpha.jar:?]
at io.airbyte.scheduler.WorkerRun.lambda$new$0(WorkerRun.java:53) [io.airbyte-airbyte-scheduler-0.14.1-alpha.jar:?]
at io.airbyte.scheduler.WorkerRun.call(WorkerRun.java:61) [io.airbyte-airbyte-scheduler-0.14.1-alpha.jar:?]
at io.airbyte.scheduler.WorkerRun.call(WorkerRun.java:42) [io.airbyte-airbyte-scheduler-0.14.1-alpha.jar:?]
at io.airbyte.commons.concurrency.LifecycledCallable.execute(LifecycledCallable.java:114) [io.airbyte-airbyte-commons-0.14.1-alpha.jar:?]
at io.airbyte.commons.concurrency.LifecycledCallable.call(LifecycledCallable.java:98) [io.airbyte-airbyte-commons-0.14.1-alpha.jar:?]
at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
at java.lang.Thread.run(Thread.java:832) [?:?]
Suppressed: io.airbyte.workers.WorkerException: Tap process wasn't successful
at io.airbyte.workers.normalization.DefaultNormalizationRunner.close(DefaultNormalizationRunner.java:100) ~[io.airbyte-airbyte-workers-0.14.1-alpha.jar:?]
at io.airbyte.workers.DefaultSyncWorker.run(DefaultSyncWorker.java:109) [io.airbyte-airbyte-workers-0.14.1-alpha.jar:?]
at io.airbyte.workers.DefaultSyncWorker.run(DefaultSyncWorker.java:48) [io.airbyte-airbyte-workers-0.14.1-alpha.jar:?]
at io.airbyte.workers.wrappers.OutputConvertingWorker.run(OutputConvertingWorker.java:44) [io.airbyte-airbyte-workers-0.14.1-alpha.jar:?]
at io.airbyte.workers.wrappers.JobOutputSyncWorker.run(JobOutputSyncWorker.java:32) [io.airbyte-airbyte-workers-0.14.1-alpha.jar:?]
at io.airbyte.scheduler.WorkerRun.lambda$new$0(WorkerRun.java:53) [io.airbyte-airbyte-scheduler-0.14.1-alpha.jar:?]
at io.airbyte.scheduler.WorkerRun.call(WorkerRun.java:61) [io.airbyte-airbyte-scheduler-0.14.1-alpha.jar:?]
at io.airbyte.scheduler.WorkerRun.call(WorkerRun.java:42) [io.airbyte-airbyte-scheduler-0.14.1-alpha.jar:?]
at io.airbyte.commons.concurrency.LifecycledCallable.execute(LifecycledCallable.java:114) [io.airbyte-airbyte-commons-0.14.1-alpha.jar:?]
at io.airbyte.commons.concurrency.LifecycledCallable.call(LifecycledCallable.java:98) [io.airbyte-airbyte-commons-0.14.1-alpha.jar:?]
at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
at java.lang.Thread.run(Thread.java:832) [?:?]
```

## Steps to Reproduce
TODO

## Severity of the bug for you
High -- this prevents the user from syncing Shopify data

**Comment by ChristopheDuong** [comment]
Nested (or arrays) columns are not supported by normalization as reported here: #886

**Comment by sherifnada** [comment]
@ChristopheDuong is this a nesting issue? The metafields stream doesn't seem to have any nested columns (at least when I try it locally)

**Comment by ChristopheDuong** [comment]
Array types are not nested columns, you are right!

In my mind, nested and arrays are treated similarly by normalization though... Going through the `extract_nested_properties` functions to extract in a separate table and I am not so sure if we happen to be blocked by the same FE/BE issue as nesting for this or not...

**Comment by sherifnada** [comment]
Confirmed with user that this issue was happening only on 0.1.6 and was fixed in 0.1.7. Closing
